pydantic/fields.py

```diff
@@ -307,8 +321,9 @@ def prepare(self) -> None:
         Note: this method is **not** idempotent (because _type_analysis is not idempotent),
         e.g. calling it it multiple times may modify the field and configure it incorrectly.
         """
-        if self.default is not None and self.type_ is None:
-            self.type_ = type(self.default)
+        default_value = self.default_factory() if self.default_factory is not None else self.default
```

Comment:
perhaps useful to have a `get_default()` function on this class which is used here and in `main.py` and perhaps deals with deepcopy?